### DISCUSSION TITLE
FMU Export Problem: Failed to load shared library

### DISCUSSION THREAD

**Discussion by Modelica1212** [OpenModelica/OpenModelica#12229] [Q&A] [answered]
Hello everyone,

I am currently exploring the possibilities of Modelica, aiming to export a small project as an FMU (Functional Mock-up Unit) and integrate it into a Python code using the FMPY Library.

I have attempted both the GUI and the Webapp. However, when trying to read a file from OpenModelica, I consistently encounter an issue resulting in the error message: "Failed to load shared library C:\.....\binaries\win64\fmutest.dll. Try using the full path with constructor syntax."

I have encountered FMU files online that successfully work with the FMPY library, but those exported from OpenModelica seem to present a problem.i added added the follwing compiler flag `--fmuRuntimeDepends=modelica `  in `Tools->Options->Simulation->Additonal Translation Flags- ` as mentionned in [https://stackoverflow.com/questions/78019725/fmu-export-from-openmodelica-to-python-fmpy#:~:text=Workaround%20Set%20compiler%20flag%20--fmuRuntimeDepends%3Dmodelica%20%28Tools-%3EOptions-%3ESimulation-%3EAdditonal%20Translation%20Flags%29,to%20the%20latest%20release%20or%20the%20nightly%20version.] but it is still the same Problem.

Is there a specific setting to change or a particular requirement in the code to make it function properly? Any insights or suggestions would be greatly appreciated.

Below is the code I've used as an example:

` model fmutest
 parameter  Real a=1;
output Real c;
Real b;
equation
b=2* time ;
c=a*sin(b);
end fmutest; `
The FMU file is attached . [fmutest.zip](https://github.com/OpenModelica/OpenModelica/files/14957417/fmutest.zip)
OpenModelica v1.23.0-dev
thank you in advance 

Regards

**Comment by casella** [comment]
@Modelica1212 we are currently testing FMU-ME export and simulation of the FMUs with FMPy in our CI, see the reports linked in [this page](https://libraries.openmodelica.org/branches/master-fmi-fmpy/). The success rate is still not as good as with the regular simulation runtime, but it's encouraging. For example, when trying the model of the [Buildings](https://libraries.openmodelica.org/branches/master-fmi-fmpy/Buildings_latest/Buildings_latest.html) library (which is quite challenging), we have a simulation success rate of 86% of the models in the library.

These tests are performed on Linux. We are also occasionally testing them on Windows. Unfortunately, there are a number of things that can go wrong with other operating systems and/or with specific installations, and we are trying to figure them all out and fix them. Welcome to DLL hell 😅 

Can you please try with the very latest 1.24.0-dev nightly build, before we dig further? Thanks!

**Reply by AnHeuermann** [reply, marked as answer]
Are you on Windows 10? Probably related to https://github.com/OpenModelica/OpenModelica/issues/12065.
After using the latest OpenModelica try switching the C compiler from clang to gcc:

![grafik](https://github.com/OpenModelica/OpenModelica/assets/38031952/150b155f-8043-4b51-b212-a03b576bc82a)

**Comment by Modelica1212** [comment]
Hello, after switching to the C compiler, the problem seems to be solved, and it is working with my current simulations. Thank you very much for your help.
Regards

**Reply by casella** [reply]
OK, good. using gcc instead of clang will take a bit more time to generate the FMU, but other than that it will be OK. Unfortunately this is very much system-specific, on my Windows 10 PC I can successfully export FMUs with clang and it's still not entirely clear why it sometimes work and sometimes not.

**Comment by sergiochavezlazo96** [comment]
Hello, I am using OpenModelica v1.23.0 (64 bit) in Windows 10 to export an FMU for model exchange. Currently I am able to generate it and test it using fmpy. However, when I share this FMU with someone else using Windows 11, I am getting the same error mentioned here.

I tried to export my model again using gcc instead of clang but is still failing in his computer

How can I fix this? We are both using a system type x64

**Reply by AnHeuermann** [reply]
Please try again with the latest [nightly build](https://openmodelica.org/download/download-windows/). If the issue is still there describe how to reproduce it. And add the full error message to your post. If we can reproduce the issue we can fix it most likely.
